doc/arch/adr-002-rummager-integration-via-publishing-api.md

```diff
@@ -0,0 +1,45 @@
+# ADR 2: Rummager integration via Publishing API
+
+## Context
+
+We are currently moving to use the publishing API as the source of truth for data in our search index - we had each application integrate directly with rummager - this will help standardise how data is inserted into our index and simplify the reindexing process.
+
+Smart answers currently sends a single document to rummager per a smart answer which is indexed under the smart page URL. This document contains all the text from all the nodes (questions and outcomes) associated with the smart answer.
+
+In addition to this smart answers sends two documents to the publishing API for each smart answer:
+
+1. A document representing the start page, this document only contains the text present on the start page. This has a document type of `transaction`
+1. A document representing the flow, this has the URL for the flow base page and does not contain any content. This has a document type of `smart-answer`
+
+## Options
+
+In order to move to indexing via rummager we will need to change what data is sent to the publishing API, the following options have been considered:
+
+1. Add the content for each node to a new array item on the document schema at `details -> nodes` for the start page.
+
+  This has the negative side effect of adding the new item to all `transaction` dcument types - this document type is also used by the `publisher` application. It has also been suggested that "something feels wrong about adding context-specific data to a schema because search happens to use it".
+
+1. Change the document type for the start page then add the new array item to the new content schema.
+
+  This would avoid the issue posed by multiple applications using the same document type, however this is potentially a large piece of work.
+
+1. Add the content to any existing document schema item at `hidden_indeaxable_content` for the flow base page.
+
+  This would increase the system complexity as it would require rummager to merge the two records from the publishing API into a single item in the rummager index, this functionality is not currently supported and would require a reworking of the queue based processing.
+
+  This raises the question about the `hidden_indexable_content` field which is something that we want to remove in the future as it is considered a hack which is primarily used by `specialist_publisher` to store PDF content as we are not currently able to extract it from the PDF documents.
+
+1. Add the content to any existing document schema item at `hidden_indeaxable_content` for the start page.
+
+  As stated above we want to remove this field long term.
+
+  This is consider the easiest option to implement.
```

Comment:
I agree with this, it seems like the most pragmatic option for the short term out of the ones listed. However I feel that `hidden_indexable_content` doesn't go far enough to separate data from presentation, so this just clutters up the API with fields that cannot be reused for anything else.

I think it's a bad idea to couple individual fields in the content store/API to rummager fields. The stuff that smart answers would call `hidden_indexable_content` feels different from the content specialist publisher considers `hidden_indexable_content`, and both are very different from the real `indexable_content`, which is content that is immediately available on the page without clicking through to stuff. It's also possible that the way we represent this information in rummager could change in the future in order to better balance the influences on search results.

Reply:
I think it's a lot better to create a field that explicitly contains the extra stuff for rummager than to try to put this content into a schema structure that can't actually be used. I understand the desire not to couple, but I think it's probably unavoidable for publishing apps to send search-specific data to publishing-api. The alternative is to clutter the system with schema / schema changes that secretly are rummager-only, but look generic (https://github.com/alphagov/govuk-content-schemas/pull/677 is another example).